lib/uktest/Makefile.uk

```diff
@@ -0,0 +1,12 @@
+$(eval $(call addlib_s,libuktest,$(CONFIG_LIBUKTEST)))
+
+CINCLUDES-$(CONFIG_LIBUKTEST)   += -I$(LIBUKTEST_BASE)/include
+CXXINCLUDES-$(CONFIG_LIBUKTEST) += -I$(LIBUKTEST_BASE)/include
+
+LIBUKTEST_SUPPRESS_FLAGS-y   += -Wno-unused-function
+LIBUKTEST_SUPPRESS_FLAGS-y   += -Wno-unused-parameter
+LIBUKTEST_CFLAGS-y           += $(LIBUKTEST_SUPPRESS_FLAGS-y)
```

Comment:
```suggestion
```

It is bad style to add suppression flags for libraries that we are writing and where the warnings can be selectively suppressed via attributes. This ensures that we do handle the warning in cases, were we later add code and really do a mistake. See suggested attribute for `_uk_test_do_assert`